internal/compiler/widgets/fluent-base/std-widgets-impl.slint

```diff
@@ -261,5 +261,124 @@ export component ScrollView {
         width: 16px;
         background: root.enabled ? transparent : Palette.neutralLighter;
     }
+}
+
+export component Switch {
+    callback toggled;
+
+    in property <string> text <=> label.text;
+    in-out property <bool> checked: true;
+    out property <bool> has-focus: fs.has-focus;
+    in property<bool> enabled: false;
+
+    min-width: 40px;
+    min-height: 20px;
+    vertical-stretch: 0;
+    horizontal-stretch: 0;
+    accessible-label <=> label.text;
+    accessible-checkable: true;
+    accessible-checked <=> root.checked;
+    accessible-role: checkbox;
+
+    layout := HorizontalLayout {
+        spacing: 8px;
+
+        VerticalLayout {
+            alignment: center;
+
+           Rectangle {
+                width: 40px;
+                height: 20px;
+
+                track := Rectangle {
+                    border-radius: 10px;
+                    border-width: 1px;
+                    border-color: Palette.neutralSecondaryAlt;
+                }
+
+                handle := Rectangle {
+                    x: 3px;
+                    y: (parent.height - self.height) / 2;
+                    width: 14px;
+                    height: self.width;
+                    border-radius: self.width / 2;
+                    background: Palette.neutralSecondaryAlt;
+
+                    animate background, x { duration: 200ms; easing: linear; }
+                }
+
+                animate background, border-color{ duration: 200ms; easing: linear; }
+            }
+        }
+
+        label := Text {
+            color: !root.enabled ? Palette.neutralTertiary : Palette.neutralDark;
+            horizontal-alignment: left;
+            vertical-alignment: center;
+            vertical-stretch: 1;
+        }
+    }
+
+    touch := TouchArea {
+        enabled <=> root.enabled;
+
+        clicked => {
+            root.toggle-checked();
+        }
+    }
+
+    fs := FocusScope {
```

Comment:
Any use of `FocusScope` inside one of our widgets should be accompanied with a binding to the `forward-focus` property, so that - in this case - somebody calls `some_switch.focus();`, it forwards the focus to the `FocusScope` inside. (In fact, without the `forward-focus`, it won't be possible to call `.focus()`).